py2puml/__init__.py

```diff
@@ -1,17 +1,26 @@
 from os.path import dirname, realpath, join
-from re import compile, Match
+from re import compile
+from itertools import takewhile
+
+# exports the version and the project description read from the pyproject.toml file
+__version__ = None
+__description__ = None
 
-# exports the version read from the pyproject.toml file
 try:
     PARENT_DIR = dirname(dirname(realpath(__file__)))
     VERSION_PATTERN = compile('^version = "([^"]+)"$')
-    with open(join(PARENT_DIR, 'pyproject.toml')) as pyproject:
-        version_match: Match = next((
-            match for match in (VERSION_PATTERN.search(line) for line in pyproject.readlines())
-            if match is not None
-        ), None)
+    DESCRIPTION_PATTERN = compile('^description = "([^"]+)"$')
 
-        __version__ = version_match.group(1)
-except:
-    __version__ = None
+    def get_from_line_and_pattern(line: str, pattern) -> str:
+        pattern_match = pattern.search(line)
+        if pattern_match is None:
+            return None
+        else:
+            return pattern_match.group(1)
 
+    with open(join(PARENT_DIR, 'pyproject.toml')) as pyproject:
+        for line in takewhile(lambda _: __version__ is None or __description__ is None, pyproject):
+            __version__ = __version__ if __version__ is not None else get_from_line_and_pattern(line, VERSION_PATTERN)
+            __description__ = __description__ if __description__ is not None else get_from_line_and_pattern(line, DESCRIPTION_PATTERN)
```

Comment:
It is the `pyproject.toml` parsing which causes [the problem I was describing](https://github.com/lucsorel/py2puml/pull/9#issuecomment-780608666).

Do you publish `py2pulm` on PypI using poetry as-well?

Reply:
hello,

Yes I publish this library to PyPI using poetry.

Thank you for your investigation about the plantuml server (and the API being deactivated on the public plantuml server - I believe they did it to reduce the workload it would receive otherwise). I think it would be interesting to add the docker command which runs a local plantuml server in the README.md's CLI documentation. What do you think?
 
Regarding the issue with the `pyproject.toml` parsing, I think I will move it into a test script (which runs with poetry in a development or CI/CD environment) so that the production code (and the CLI in particular) does not do it anymore. That would solve this last issue, wouldn't it?